lib/readline.js

```diff
@@ -931,6 +1019,11 @@ Interface.prototype._ttyWrite = function(s, key) {
     switch (key.name) {
       case 'return':  // carriage return, i.e. \r
         this._sawReturnAt = Date.now();
+        if (this.inReverseSearch) {
+          this.line = this.history[this.reverseSearchIndex - 1];
+          this.inReverseSearch = false;
+          this.reverseSearchIndex = 0;
+        }
```

Comment:
Good point, yes I did test on this. Actually, all the keys like `left`, `right`, `ctrl + a` are working as expected (as the whole reverse search is on the same stream and cursor set correctly). In this case, I guess it's good enough to leave these as is, rather than writing a layer which prevents all these key combinations when search is on. What say? Thoughts?